### PR TITLE
Improve ore vein API (both internal and for KubeJS)

### PR DESCRIPTION
**Pull request by mikerooni**

### CODE REVIEW

File: common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/ores/OreVeinUtil.java [hunk 1]
```diff
@@ -95,4 +110,28 @@ static int getMaxVeinSearchDistance() {
...
     }
+
+    @Nullable
+    public static Supplier<HolderSet<Biome>> resolveBiomes(List<String> biomes) {
+        if (biomes.isEmpty())
+            return null;
+
+        RegistryOps<JsonElement> registryOps = RegistryOps.create(JsonOps.INSTANCE, GTRegistries.builtinRegistry());
+        return () -> RegistryCodecs.homogeneousList(Registries.BIOME)
+                .decode(registryOps, resolveBiomeCodecInput(biomes))
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/ores/OreVeinUtil.java]: TODO: resolve this at the beginning for better errors in kjs

File: common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/IWorldGenLayer.java [hunk 2]
```diff
@@ -15,11 +18,36 @@ public interface IWorldGenLayer extends StringRepresentable {
...
+    Set<ResourceLocation> getLevels();
     RuleTest getTarget();
 
 
     @FunctionalInterface
     interface RuleTestSupplier {
         RuleTest get();
     }
+
+    static IWorldGenLayer nowhere() {
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/IWorldGenLayer.java]: TODO: singleton

File: common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/WorldGenLayers.java [hunk 3]
```diff
@@ -10,6 +10,7 @@
 import net.minecraft.tags.BlockTags;
 import net.minecraft.util.StringRepresentable;
 import net.minecraft.world.level.Level;
+import net.minecraft.world.level.levelgen.structure.templatesystem.AlwaysTrueTest;
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/WorldGenLayers.java]: TODO remove

File: common/src/main/java/com/gregtechceu/gtceu/integration/kjs/events/GTOreVeinEventJS.java [hunk 4]
```diff
@@ -14,19 +17,38 @@ public GTOreVeinEventJS() {
...
     }
 
-    public void modify(ResourceLocation id, Consumer<GTOreDefinition> consumer) {
-        consumer.accept(GTRegistries.ORE_VEINS.get(id));
+    public void removeAll() {
+        Set<ResourceLocation> keys = Set.copyOf(GTRegistries.ORE_VEINS.keys());
+        keys.forEach(this::remove);
+    }
+
+    public void removeAll(BiPredicate<ResourceLocation, GTOreDefinition> predicate) {
```
Comment (mikerooni) [inline common/src/main/java/com/gregtechceu/gtceu/integration/kjs/events/GTOreVeinEventJS.java:50]: TODO: actually use the predicate here

**Review by screret — APPROVED** [review]

**Pull request merged**